route/route.go

```diff
@@ -191,6 +199,45 @@ func (r *Route) PathSelection() {
 	r.updateEqualPathCount()
 }
 
+// Euql compares Are two routes and return true if they are equal
+func (r *Route) Equal(other *Route) bool {
+	r.mu.Lock()
+	defer r.mu.Unlock()
+
+	a := r.pfx.Equal(other.pfx)
+	b := r.ecmpPaths == other.ecmpPaths
+	c := true
+
+	if r.paths == nil && other.paths == nil {
+		c = true
+		return a && b && c
+	}
+
+	if len(r.paths) != len(other.paths) {
+		c = false
+		return a && b && c
```

Comment:
Why not just return false?

Reply:
valid point. This function was hacked in the middle of the night at 2 PM.
Will refactor this function a bit